internal/test/e2e/run.go

```diff
@@ -150,18 +163,28 @@ type instanceRunConf struct {
 	testReportFolder, branchName                                              string
 	ipPool                                                                    networkutils.IPPool
 	hardware                                                                  []*api.Hardware
+	hardwareCount                                                             int
 	bundlesOverride                                                           bool
 	testRunnerType                                                            TestRunnerType
 	testRunnerConfig                                                          TestInfraConfig
 	cleanupVms                                                                bool
 	logger                                                                    logr.Logger
 }
 
-func RunTests(conf instanceRunConf) (testInstanceID string, testCommandResult *testCommandResult, err error) {
+//nolint:gocyclo, revive // RunTests responsible launching test runner to run tests is complex.
+func RunTests(conf instanceRunConf, hardwareQueue *hardwareQueue) (testInstanceID string, testCommandResult *testCommandResult, err error) {
```

Comment:
Did we consider adding the queue/semaphor to the configuration object?

Reply:
Yes we did. But did not think that was a great idea since the configuration object already has the tinkerbell hardware and count fields needed. Since the configuration object should ideally be provider agnostic (but already has many fields particular to Tinkerbell) thought of separating it form the configuration object!

Reply:
The config object isn't provider agnostic though, and already contains functional objects like `Session`. Having a separate parameter disassociates the configuration and, from a future maintainer perspective, raises the question "Should I add it to the config object or as a parameter".

Reply:
I agree it isn't but ideally it should be since its responsible for launching all tests. And for the questions, I think it's a valid thought consideration to have and not over populate the configuration object with too many case specific parameters given that the generic functionality of that object to run tests across provider without delving into details for each provider.